lib/dispatcher.rb

```diff
@@ -28,7 +28,17 @@ def on_approval_status_change(approval)
   end
 
   def requires_approval_notice?(approval)
-    true
+    if approval.cart.getProp('origin') == 'ncr'
+      self.with_approval_notice_requirements { approval.is_final_approval? }
```

Comment:
This could turn into a rules block that we pull out of here, potentially covering all of the different clients and addressing other client-specific functions